### PR TITLE
Zip Code Validation

### PR DESCRIPTION
**Pull request by liambell-le**
### Summary
Added a new postal code input type to allow validation for either/or US/CA postal codes.

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/formatters/postal-code-generic.formatter.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        /[0-9]/
+    ];
+    allowKey(key: string, newValue: string): boolean {
+        const len = newValue.length;
+
+        return PostalCodeGenericFormatter.FILTER_REGEXS_CA[len - 1].test(newValue[len - 1]) ||
+            PostalCodeGenericFormatter.FILTER_REGEXS_US[len - 1].test(newValue[len - 1 ]);
+    }
+
+    formatValue(value: string): string {
```
Comment (kdelorey) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/formatters/postal-code-generic.formatter.ts:26]: Does this like coerce the text as you're typing or something?

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/formatters/postal-code-generic.formatter.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+export class PostalCodeGenericFormatter implements IFormatter {
+    static readonly FILTER_REGEXS_CA = [
+        /[abceghjklmnprstvxyABCEGHJKLMNPRSTVXY]/,
+        /[0-9]/,
+        /[abceghjklmnprstvwxyzABCEGHJKLMNPRSTVWXYZ]/,
+        /[0-9]/,
+        /[abceghjklmnprstvwxyzABCEGHJKLMNPRSTVWXYZ]/,
+        /[0-9]/
+    ];
+    static readonly FILTER_REGEXS_US = [
```
Comment (kdelorey) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/formatters/postal-code-generic.formatter.ts:12]: Do you need to worry about allowing valid delivery route numbers? (eg.12345-1234)

Comment (liambell-le) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/formatters/postal-code-generic.formatter.ts:12]: The requirement in the epic does not specify route. "Zip 5 (US) Postal code 6 (Canada with the a1a 1a1 format)".

Comment (mmichalek) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/formatters/postal-code-generic.formatter.ts:12]: I think we should avoid putting this validation burden on the client if possible.  My concern is that we would have to modify the angular app to take it international. Could the server specify the valid pattern(s) that the client could use?

**Review by stevencarley — APPROVED** [review]

**Pull request merged**